mlsdbtool/main.c

```diff
@@ -0,0 +1,175 @@
+/*
+  Copyright (C) 2022 Jolla Ltd.
+  Contact: Daniel Suni <daniel.suni@jolla.com>
+
+  This file is part of geoclue-mlsdb.
+
+  Geoclue-mlsdb is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License.
+*/
+
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+
+#define NEWLINE 10
+#define STDIN 0
+
+/*
+This parser will read MLS full export data in CSV format. Data files can
+be downloaded here: https://location.services.mozilla.com/downloads
+
+The reader assumes the data is numerically sorted by unique cell id and that
+only data lines (i.e. no headers) are passed. To achieve that, run:
+tail -n +2 [CSV file] | sort -t, -k2n,2 -k3n,3 -k4n,4 -k5n,5 -k1,1 | geoclue-mlsdb-tool
+or (preferrably) use the wrapper script.
+
+---
+
+The "network" is allocated as follows:
+0000000000000000000000000000000000000000000000000000000000000000
+        \____ ___/\_______ ______/\_____________ ____________/\/
+             v            v                     v              v
+          NET: 10b Area: 16 bits        Cell ID: 28 bits     Radio: 2b 
+
+MCC pos stores the position of an array in which the MCC value is stored.
+It's a hack that enables us to store 66 bits of data in a 64 bit integer.
+
+The "position" is allocated as 2 concatenated 32-bit floats
+with longitude first and then latitude.
+*/
+uint64_t network;
+uint64_t position;
+uint64_t previous = 0;
+
+// Helper function for debugging (prints out a 64b int as binary)
+void print_bin(uint64_t n) {
+    if (n > 1) {
+        print_bin(n >> 1);
+    }
+    printf("%lu", n & 1);
```

Comment:
Compiler is warning on bunch of lines in this files, this being the first instance, hinting towards %llu.

Later some similar + "...may be used uninitialized in this function". If you could aim to minimize those, ideally with no warnings.